cmd/get_color_name.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/lucasb-eyer/go-colorful"
+)
+
+type hexColor struct {
+	Name string `json:"name"`
+	Hex  string `json:"hex"`
+}
+
+// GetColorName looks up the name for a specific hex color code.
+func GetColorName(myHexColorCode string) string {
+	var colorData []hexColor
+
+	// allColorsJSONString is defined in `all-colors.go` (generated by gen.go).
+	err := json.Unmarshal([]byte(allColorsJSONString), &colorData)
+
+	if err != nil {
+		fmt.Println("error:", err)
+	}
+
+	myHashedHexColorCode := fmt.Sprintf("#%s", myHexColorCode)
+	myReferenceColor, err := colorful.Hex(myHashedHexColorCode)
+
+	if err != nil {
+		fmt.Println("error:", err)
+	}
+
+	var closestColorIndex int
+	var closestColorDistance float64 = 1.0 // Default to 1.0, so subsequent colors will be closer.
+
+	for i := range colorData {
+		// Exit early if there's an exact hex code match.
+		if colorData[i].Hex == myHashedHexColorCode {
+			closestColorIndex = i
+			break
+		}
+
+		// Otherwise, we do a color distance comparison.
+		currentColor, err := colorful.Hex(colorData[i].Hex)
+
+		if err != nil {
+			fmt.Println("error:", err)
+		}
+
+		// We use the LAB color space to find the closest color based on human perceptions.
+		currentColorDistance := myReferenceColor.DistanceLab(currentColor)
```

Comment:
Oh interesting library! Were you able to just drop this in to replace what you used in JS?

Reply:
Yeah, I was expecting to write this part from scratch until I found this library.